pydis_site/apps/api/models/bot/aoc_completionist_block.py

```diff
@@ -0,0 +1,26 @@
+from django.db import models
+
+from pydis_site.apps.api.models.bot.user import User
+from pydis_site.apps.api.models.mixins import ModelReprMixin
+
+
+class AocCompletionistBlock(ModelReprMixin, models.Model):
+    """A Discord user blocked from getting the AoC completionist Role."""
+
+    user = models.OneToOneField(
+        User,
+        on_delete=models.CASCADE,
+        help_text="The user that is blocked from getting the AoC Completionist Role",
+        primary_key=True
+    )
+
+    is_blocked = models.BooleanField(
+        default=True,
+        help_text="Whether this user is actively being blocked "
+                  "from getting the AoC Completionist Role",
+        verbose_name="Blocked"
+    )
+    reason = models.TextField(
+        null=True,
```

Comment:
Better to use `blank` instead of `null`. But maybe we should leave it `null` since that seems to be what other endpoints use for similar fields (and change them all later in a separate PR if we want)?

https://docs.djangoproject.com/en/3.2/ref/models/fields/#null

Reply:
Yes, I used Infractions as a reference, to keep things consistent